src/lib.rs

```diff
@@ -192,6 +230,34 @@ impl SecretClient {
         let auth_info = kv2::set(&self.client, &self.mount, path, data).await?;
         Ok(auth_info)
     }
+
+    #[cfg(feature = "write")]
+    pub async fn set_secrets_with_options<A>(
+        &self,
+        options: WriteSecretOptions<'_, A>,
+    ) -> Result<SecretVersionMetadata>
+    where
+        A: Serialize,
+    {
+        use vaultrs::api::kv2::requests::SetSecretRequestOptions;
+        let path = options.path.unwrap_or(&self.base_path);
```

Comment:
Wdyt about `unwrap_or_else` to be lazy?

Reply:
:+1:, this would be more of a "idiomatic rust" thing? I find `.unwrap_or_else (|| ...)` has more "syntactical noise" and in this case we are not really gaining something practical from the lazy-ness right?

Reply:
Not really, it is either a reference or a closure 🤷🏾‍♂️ . Fine with both, I would do it for the sake of clean convention. :trollface: